frontend/app/pages/admin/admin.tsx

```diff
@@ -0,0 +1,600 @@
+    {
+      id: "actions",
+      cell: ({ row }) => {
+        const index = row.index;
+        return (
+          <DropdownMenu>
+            <DropdownMenuTrigger asChild>
+              <Button variant="ghost" className="h-8 w-8 p-0">
+                <span className="sr-only">Open menu</span>
+                <MoreHorizontal className="h-4 w-4" />
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent align="end">
+              <DropdownMenuItem onClick={() => onEdit(index)}>
+                Edit Subscription
+              </DropdownMenuItem>
+              <DropdownMenuItem
+                onClick={() => onDelete(index)}
+                className="text-red-500"
+              >
+                Delete Subscription
+              </DropdownMenuItem>
+            </DropdownMenuContent>
+          </DropdownMenu>
+        );
+      },
+    },
+  ];
+}
+
+export function getUserColumns(
+  onRoleChange: (index: number, role: "admin" | "user") => void,
+  onDelete: (index: number) => void,
+): ColumnDef<User>[] {
+  return [
+    {
+      accessorKey: "name",
+      header: "User",
+    },
+    {
+      accessorKey: "role",
+      header: "Role",
+      cell: ({ row }) => {
+        const role = row.getValue("role") as "admin" | "user";
+        const index = row.index;
+
+        return (
+          <Select
+            value={role}
+            onValueChange={(newRole) =>
+              onRoleChange(index, newRole as "admin" | "user")
+            }
+          >
+            <SelectTrigger className="w-[120px]">
+              <SelectValue placeholder="Select role" />
+            </SelectTrigger>
+            <SelectContent>
+              <SelectItem value="admin">admin</SelectItem>
+              <SelectItem value="user">user</SelectItem>
+            </SelectContent>
+          </Select>
+        );
+      },
+    },
+    {
+      id: "actions",
+      cell: ({ row }) => {
+        const index = row.index;
+        return (
+          <AlertDialog>
+            <AlertDialogTrigger asChild>
+              <Button variant={"ghost"}>
+                <Trash className="text-red-500" />
+              </Button>
+            </AlertDialogTrigger>
+            <AlertDialogContent>
+              <AlertDialogHeader>
+                <AlertDialogTitle>Are you sure?</AlertDialogTitle>
+                <AlertDialogDescription>
+                  This will permanently remove the user from this organization.
+                </AlertDialogDescription>
+              </AlertDialogHeader>
+              <AlertDialogFooter>
+                <AlertDialogCancel>Cancel</AlertDialogCancel>
+                <AlertDialogAction onClick={() => onDelete(index)}>
+                  Delete
+                </AlertDialogAction>
+              </AlertDialogFooter>
+            </AlertDialogContent>
+          </AlertDialog>
+        );
+      },
+    },
+  ];
+}
+export function Admin() {
```

Comment:
```suggestion
export default function Admin() {
```
to fix react router issues